# DRUID
DRUID: deep relatedness inference utilizing identity by descent

Updated 1/4/2018
version 0.9.1a

This method is currently in beta. If you find any bugs, please report to:
monica.ramstetter@gmail.com


USAGE:

DRUID requires a minimum of three files:
1) .map file (PLINK format) containing cM positions for SNPs (see https://www.cog-genomics.org/plink2/formats#map for formatting)
2) .seg file containing pairwise IBD1/2 segment information: ind1 ind2 chr IBD1/2 IBD_segment_start IBD_segment_end (see below for creation)
3) .ibd12 file containing pairwise proportion of genome shared IBD information: ind1 ind2 IBD1_proportion IBD2_proportion (see below for creation)



FILES:

DRUID requires as input pairwise IBD segment information. Although one could use IBD information generated by any pairwise IBD segment detection method (assuming the files are properly formated), we recommend using Refined IBD (Beagle 4.1: https://faculty.washington.edu/browning/beagle/beagle.html). The authors of Beagle 4.1 recommend running Refined IBD three time with different random seeds. 

We have a script to merge IBD information from the three separate Refined IBD runs and extract pairwise IBD1/2 information as well as pairwise IBD1/2 segment information. This script is available at: 
https://github.com/MonicaRamstetter/bakeoff. 

To obtain the files necessary for running DRUID, one should run the getIBD.py script as follows:

python getIBD.py -f [file1.ibd] [file2.ibd] [file3.ibd] -m [mapfile.map] -t 1 -s 1 -o [output_prefix]

where file1.ibd, file2.ibd, and file3.ibd are the three .ibd files output by Refined IBD, mapfile.map is a PLINK formatted map file containing genetic position information and corresponding to the VCF file input to Refined IBD, '-t 1' ensures pairwise proportions of genome shared IBD are output, and '-s 1' ensures segment information is output.


